extensions/cloud/portability-cloud-google/src/main/java/org/datatransferproject/cloud/google/GoogleCloudIdempotentImportExecutor.java

```diff
@@ -0,0 +1,233 @@
+package org.datatransferproject.cloud.google;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.google.cloud.datastore.*;
+import com.google.cloud.datastore.StructuredQuery.CompositeFilter;
+import com.google.cloud.datastore.StructuredQuery.PropertyFilter;
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.datatransferproject.api.launcher.Monitor;
+import org.datatransferproject.spi.transfer.idempotentexecutor.IdempotentImportExecutor;
+import org.datatransferproject.types.transfer.errors.ErrorDetail;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.Callable;
+
+import static java.lang.String.format;
+
+public class GoogleCloudIdempotentImportExecutor implements IdempotentImportExecutor {
+
+  private static final String IDEMPOTENT_RESULTS_KIND = "IdempotentResults";
+  private static final String IDEMPONTENT_ERRORS_KIND = "IdempotentErrors";
+
+  private static final String RESULTS_FIELD = "result_details";
+  private static final String JOB_ID_FIELD = "job_id";
+  private static final String IDEMPOTENT_ID_FIELD = "idempotent_id";
+  private static final String ERROR_FIELD = "error_details";
+
+  private final Datastore datastore;
+  private final Monitor monitor;
+
+  private Map<String, Serializable> knownValues;
+  private Map<String, ErrorDetail> errors;
+  private UUID jobId;
```

Comment:
Let's comment on when this is intialized

Reply:
Done